### DISCUSSION TITLE
How to deal with "Direct construction of _pytest.python.Function has been deprecated, please use _pytest.python.Function.from_parent."

### DISCUSSION THREAD

**Discussion by ebeffim** [pytest-dev/pytest#10951] [Q&A] [answered]
While updating python to 3.10, we also updated pytest and plugins to the latest version:
pytest==7.3.1
pytest-cov==4.0.0
pytest-quickcheck==0.9.0
pytest-rerunfailures==11.1.2
pytest-watch==4.2.0
pytest-timeout==2.1.0

After the update, all tests fail with this error
_______ ERROR collecting tests/integration/xxx.py ________
Direct construction of _pytest.python.Function has been deprecated, please use _pytest.python.Function.from_parent.
See https://docs.pytest.org/en/stable/deprecations.html#node-construction-changed-to-node-from-parent for more details.

This seems to be a pytest internal issue because there are no other exceptions and errors on our test code. 

How can we resolve this issue? Thoughts and comments are welcome

**Comment by nicoddemus** [comment, marked as answer]
Hi,

Run with `--fulltrace` so we can get the full traceback, and pinpoint the code which is using the deprecated way of creating functions.

**Reply by ebeffim** [reply]
--full-trace does help to pin down the issue. The root cause is actually in our source code.  Our implementation of pytest_pycollect_makeitem() does include .Function().
